### ISSUE TITLE
setup.dashboard.index should respect `elasticsearch.output.index`

### ISSUE THREAD

**Issue by jordansissel** [elastic/beats#6926]
Tested on auditbeat 6.2.4

When setting `elasticsearch.output.index`, the default `setup.dashboard.index` pattern that gets used is still `auditbeat-*`.

Here's the scenario I went through:

When changing the index name:

```
output.elasticsearch:
  index: "foo"
```

Beats setup will tell me:

> Exiting: setup.template.name and setup.template.pattern have to be set if index name is modified.

So I follow the instructions:

```
setup:
  template.name: "auditbeat"
  template.pattern: "*-auditbeat-*"
```

Then run setup again:

```
# auditbeat setup -c /tmp/auditbeat.yml -e
...
2018-04-23T23:10:09.706Z        INFO    instance/beat.go:583    Kibana dashboards successfully loaded.
```

However, when viewing the dashboards, all the dashboards/visuals appear to point at `auditbeat-*` instead of my configured `*-auditbeat-*`.

To work around this, I added:

```
setup:
  template.name: "auditbeat"
  template.pattern: "*-auditbeat-*"
  template.enabled: true
  dashboards.index: "*-auditbeat-*"   # this is required.
```

**Comment by RealLinkers** [comment]
This is still a thing on 7.2.0, I am unable to populate the kibana custom index pattern, although it does show the correct index on visualization.

**Comment by pdubois** [comment]
I saw that customer  memory to 2GB then you shrank it bank to 1GB on  kibana node.
On our cloud implementation nodes are running on allocators. The allocator kibana was initially on was i-032281cb4e2d5dca5, it was big (121GB) very
full and many small nodes running on it.
The allocator you are on now is  i-00c99f97995cc56fb it is smaller (27GB) and much less nodes running on it.
The side effect of the operation grow/shrink was to move your kibana node from one allocator to another, therefore I believe that there was a problem on the initial allocator but we are not sure of that?
I think you also performed other operations like growing and shrinking other nodes, like APM node and Elastic node.

**Comment by kvch** [comment]
Fixed by #17749 
Feel free to reopen if the problem persist on newer versions.
